tasks/ubuntu_repositories.yml

```diff
@@ -1,6 +1,18 @@
 ---
 - name: Add Ubuntu repositories
+  become: true
+  # Use the builtin Ubuntu capability for adding universe, within which wireguard lives
+  command: "add-apt-repository universe"
```

Comment:
Is there a way to do this using `apt_repo` module?

Reply:
Sadly no - the apt_repo module doesn't support this.

Reply:
Oh - apologies, you mean apt_repo, not apt_repository.  Without looking too deeply, this might be possible, but do you wish to:

1. Include a dependency on https://galaxy.ansible.com/community/general
2. By extension, cause the apt_repo collection to install the `apt-repo` CLI, which isn't included in Ubuntu by default...

It feels a little convoluted!

Reply:
The apt-repository -compatible approach would be to (re-)add the universe repository (with a generated URL) regardless of whether it's already there.  The challenge, unfortunately, is mirror selection.  Generally Ubuntu systems select a nearby deb mirror when installed... so generating the URL will be tricky.

add-apt-repository does this logic for us, by examining the content of sources.list and adding the requested source iff it's not already listed.  It's pretty neat.

Reply:
So this is a little stream-of-consciousness, but looking at this:
https://unix.stackexchange.com/questions/43165/how-does-apt-get-deal-with-multiple-sources

I think I'll use apt-repository to add a `sources.list.d/ansible-wireguard` generated universe source.  If universe is already included in sources.list, it'll be used preferentially, and if not, the sources.list.d file will be used.

I'll put another change in for this.

Reply:
... stream of consciousness done, apologies.  Above is my last word on the subject!

Reply:
Thank you for the elaborate answer. Let's keep things the way they are for the sake of simplicity